### PR TITLE
Factor usage global.param state into separate struct

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Comment by ibuclaw** [comment]
This doesn't need to be in globals. Only the main modules use this.

**Comment by thewilsonator** [comment]
One step at a time, feel free to follow up on this.

**Comment by ibuclaw** [comment]
Previous discussion  #13795.

**Review by dkorpel — APPROVED** [review]

File: compiler/src/dmd/globals.d [hunk 1]
```diff
@@ -101,6 +101,23 @@ extern(C++) struct Output
     OutBuffer* buffer;  // if this output is buffered, this is the buffer
     int bufferLines;    // number of lines written to the buffer
 }
+
+/// Command line state related to printing uasage about other switches
```
Comment (tgehr) [inline compiler/src/dmd/globals.d:105]: Typo: `uasage`

Comment (thewilsonator) [reply compiler/src/dmd/globals.d]: Fixed, thanks!

**Pull request merged**